core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -34,6 +34,7 @@ import com.fasterxml.jackson.databind.annotation.JsonDeserialize
 import org.apache.hadoop.fs.{FileStatus, FileSystem, Path, SafeModeAction}
 import org.apache.hadoop.hdfs.DistributedFileSystem
 import org.apache.hadoop.security.AccessControlException
+import org.apache.hadoop.util.BlockingThreadPoolExecutorService
```

Comment:
this class is marked with `@Private`, but the API has been stable since 2016, so it should be low risk to use directly.

Reply:
Well, I'm not sure this is a good decision.

Reply:
I understand the reasoning. However, I disagree with the point of cutting off the Hadoop deps. If that's tricky now, it will be the same in the end.

Without moving forward, we need to check the following.
1. Can we borrow `Private`-scope code in Spark side stably without increasing further maintenance burden?
2. Can we ask Apache Hadoop community to expose that class officially? This will remove our forked code at that time.

Historically, we spent lots of efforts to migrate to `Hadoop Client` to avoid this kind of dependencies issue. I don't think both the Apache Spark and Apache Hadoop community wants to have a kind of hard dependencies again.

Reply:
@dongjoon-hyun @peter-toth The Hadoop code comes from [Apache S4](https://github.com/apache/incubator-retired-s4/blob/0.6.0-Final/subprojects/s4-comm/src/main/java/org/apache/s4/comm/staging/BlockingThreadPoolExecutorService.java), I wrote a simple "BlockingThreadPoolExecutorService" based on it, so we don't have to worry about using Hadoop private API.

Reply:
BTW, I also verified the updated code on the prod env, and it has the same effect as the first version.